rotkehlchen/chain/bitcoin/types.py

```diff
@@ -0,0 +1,111 @@
+import logging
+from dataclasses import dataclass
+from enum import Enum, auto
+from typing import Any, NamedTuple
+
+from rotkehlchen.errors.serialization import DeserializationError
+from rotkehlchen.fval import FVal
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.serialization.deserialize import deserialize_timestamp
+from rotkehlchen.types import BTCAddress, TimestampMS
+from rotkehlchen.utils.misc import satoshis_to_btc, ts_sec_to_ms
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class BtcScriptType(Enum):
+    """Represents the type of a Bitcoin script.
+    See https://learnmeabitcoin.com/technical/script/#standard-scripts
+    """
+    # Pay-to-Public-Key (Legacy) - lock output to public key - the simpler version of p2pkh
+    P2PK = auto()
+    # Pay-to-Public-Key-Hash (Legacy) - lock output to public key hash (send somebody btc)
+    P2PKH = auto()
+    # Pay-to-MultiSig (Legacy) - lock btc to multiple public keys
+    P2MS = auto()
+    # Pay-to-Script-Hash (Legacy) - lock btc to the hash of a script
+    P2SH = auto()
+    # lock data in the transaction - Used to store data on the blockchain
+    OP_RETURN = auto()
+    # Pay-to-Witness-Public-Key-Hash - segwit lock to public key hash (similar to p2pkh)
+    P2WPKH = auto()
+    # Pay-to-Witness-Script-Hash - seqwit lock to hash of script (similar to p2wsh)
+    P2WSH = auto()
+    # Pay-to-Taproot - lock output to a public key and multiple optional custom locking scripts
+    P2TR = auto()
+    # Other non-standard script
+    CUSTOM = auto()
+
+
+API_SCRIPT_TYPE_MAPPING = {
+    'p2pk': BtcScriptType.P2PK,
+    'p2pkh': BtcScriptType.P2PKH,
+    'p2ms': BtcScriptType.P2MS,
+    'p2sh': BtcScriptType.P2SH,
+    'op_return': BtcScriptType.OP_RETURN,
+    'v0_p2wpkh': BtcScriptType.P2WPKH,
+    'v0_p2wsh': BtcScriptType.P2WSH,
+    'v1_p2tr': BtcScriptType.P2TR,
+}
+
+
+class TxInOut(NamedTuple):
+    """Represents an individual input/output of a Bitcoin transaction."""
+    value: FVal
+    type: BtcScriptType
+    raw_script: str
+    address: BTCAddress | None
+    is_out: bool
+
+    @classmethod
+    def deserialize(cls, data: dict[str, Any], is_out: bool) -> 'TxInOut':
+        try:
+            if (raw_type := data.get('scriptpubkey_type')) is None or raw_type not in API_SCRIPT_TYPE_MAPPING:
```

Comment:
a question. `vin` will never have `scriptpubkey_type` as seen in the screenshot here right?
![CleanShot 2025-06-13 at 11 58 10@2x](https://github.com/user-attachments/assets/ff332b08-89c1-4934-9389-1a7480c15520)


so, i guess you need to check the `prevout` for `vin`s